lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -2769,24 +2788,18 @@ constructTransaction txLayer netLayer db wid txCtx preSel = do
     mkUnsignedTransaction txLayer xpub pp txCtx (Left preSel)
         & withExceptT ErrConstructTxBody . except
 
--- | Construct an unsigned transaction from a given selection
--- for a shared wallet.
-constructSharedTransaction
-    :: forall ctx s k (n :: NetworkDiscriminant) era.
-        ( HasTransactionLayer k 'CredFromScriptK ctx
-        , HasDBLayer IO s k ctx
-        , HasNetworkLayer IO ctx
-        , k ~ SharedKey
-        , s ~ SharedState n k
-        , Typeable n
-        , WriteTx.IsRecentEra era
-        )
-    => ctx
+constructUnbalancedSharedTransaction
+    :: forall (n :: NetworkDiscriminant) ktype era
+     . (WriteTx.IsRecentEra era, Typeable n)
+    => TransactionLayer SharedKey ktype SealedTx
+    -> NetworkLayer IO Block
+    -> DBLayer IO (SharedState n SharedKey) SharedKey
     -> WalletId
     -> TransactionCtx
-    -> SelectionOf TxOut
-    -> ExceptT ErrConstructTx IO (Cardano.TxBody era)
-constructSharedTransaction ctx wid txCtx sel = db & \DBLayer{..} -> do
+    -> PreSelection
+    -> ExceptT ErrConstructTx IO
+    (Cardano.TxBody era, Maybe ([(TxIn, TxOut)] -> Map TxIn (CA.Script KeyHash)) )
```

Comment:
```
Maybe ([(TxIn, TxOut)] -> Map TxIn (CA.Script KeyHash))
```
This could be
```
Maybe ([(TxIn, TxOut)] -> [CA.Script KeyHash])
```
right? as we call `Map.elements` on it anyway in `balanceTransaction`.

Reply:
sounds reasonable -> DONE Although we will need more sophisticated script resolvers soon